docs/advanced-features/middleware.md

```diff
@@ -47,7 +48,7 @@ export function middleware(request: NextRequest) {
 }
 
 // See "Matching Paths" below to learn more
-export const config = {
+export const config: PageConfig = {
```

Comment:
This is invalid and `matcher` isn't exposed on the `PageConfig` type x-ref: https://github.com/vercel/next.js/blob/ff19e7ad304fe10c05b67b0e1eb26cd32de00c18/packages/next/types/index.d.ts#L83

Reply:
Sorry! It looks like middleware is using [MiddlewareConfig](https://github.com/vercel/next.js/blob/canary/packages/next/src/build/analysis/get-page-static-info.ts), but its type has all properties marked as required, which makes it kinda unusable in actual middleware file.

Reply:
`MiddlewareConfig` is an internal type, a separate public type should be exposed instead.

Reply:
Alright. I have renamed `MiddlewareConfig` to `InternalMiddlewareConfig`, added new `MiddlewareConfig` near `PageConfig`, and written some documentation.